nuxeo-jsf/nuxeo-platform-webapp-base/src/main/java/org/nuxeo/ecm/webapp/clipboard/ClipboardActionsBean.java

```diff
@@ -871,34 +922,29 @@ public boolean isCacheEnabledForSelection() {
     @Override
     public String exportWorklistAsZip(List<DocumentModel> documents, boolean exportAllBlobs) {
         try {
-            FacesContext context = FacesContext.getCurrentInstance();
+            DownloadService dnd = Framework.getService(DownloadService.class);
             DocumentListZipExporter zipExporter = new DocumentListZipExporter();
             File tmpFile = zipExporter.exportWorklistAsZip(documents, documentManager, exportAllBlobs);
             if (tmpFile == null) {
                 // empty zip file, do nothing
                 facesMessages.add(StatusMessage.Severity.INFO, messages.get("label.clipboard.emptyDocuments"));
                 return null;
-            } else {
-                if (tmpFile.length() > Functions.getBigFileSizeLimit()) {
-                    HttpServletRequest request = (HttpServletRequest) context.getExternalContext().getRequest();
-                    request.setAttribute(NXAuthConstants.DISABLE_REDIRECT_REQUEST_KEY, true);
-                    String zipDownloadURL = BaseURL.getBaseURL(request);
-                    zipDownloadURL += DownloadService.NXBIGZIPFILE + "/";
-                    zipDownloadURL += tmpFile.getName();
-                    try {
-                        context.getExternalContext().redirect(zipDownloadURL);
-                    } catch (IOException e) {
-                        log.error("Error while redirecting for big file downloader", e);
-                    }
-                } else {
-                    ComponentUtils.downloadFile(tmpFile, "clipboard.zip", "clipboardZip");
-                    tmpFile.delete();
-                }
-
+            }
+            try {
+                Blob blob = new FileBlob("zip");
```

Comment:
Why do you need to create a new temporary file, we already have one? If it's because we only have a `File` and not a `Blob` which is what you need, then it's better to change `DocumentListZipExporter.exportWorklistAsZip` to return a temporary `Blob` instead and cleanup the temporary file management (no `tmpFile.deleteOnExit()` for instance).